rocAL/rocAL_pybind/amd/rocal/decoders.py

```diff
@@ -0,0 +1,244 @@
+import amd.rocal.types as types
+import rali_pybind as b
+from amd.rocal.pipeline import Pipeline
+
+def image(*inputs, user_feature_key_map = None, path='', file_root ='', annotations_file= '', shard_id = 0, num_shards = 1, random_shuffle = False, affine=True, bytes_per_sample_hint=0, cache_batch_copy= True, cache_debug = False, cache_size = 0, cache_threshold = 0,
+                 cache_type='', device_memory_padding=16777216, host_memory_padding=8388608, hybrid_huffman_threshold= 1000000, output_type = types.RGB,
+                 preserve=False, seed=-1, split_stages=False, use_chunk_allocator= False, use_fast_idct = False, device = None):
+    reader = Pipeline._current_pipeline._reader
+    if( reader == 'COCOReader'):
+        kwargs_pybind = {
+            "source_path": file_root,
+            "json_path": annotations_file,
+            "color_format": output_type,
+            "shard_id": shard_id,
+            "num_shards": num_shards,
+            'is_output': False,
+            "shuffle": random_shuffle,
+            "loop": False,
+            "decode_size_policy": types.MAX_SIZE,
+            "max_width": 0,
+            "max_height":0}
+        decoded_image = b.COCO_ImageDecoderShard(Pipeline._current_pipeline._handle ,*(kwargs_pybind.values()))
+
+    elif (reader == "TFRecordReaderClassification" or reader == "TFRecordReaderDetection"):
+        kwargs_pybind = {
+            "source_path": path,
+            "color_format": output_type,
+            "num_shards": num_shards,
+            'is_output': False,
+            "user_key_for_encoded": user_feature_key_map["image/encoded"],
+            "user_key_for_filename": user_feature_key_map["image/filename"],
+            "shuffle": random_shuffle,
+            "loop": False,
+            "decode_size_policy": types.USER_GIVEN_SIZE,
+            "max_width": 2000,
```

Comment:
Looks like only TFRecordReaderDetection uses user_size?

Reply:
Yes Rajy

Reply:
Do you know why?